examples/dictation/index.html

```diff
@@ -0,0 +1,105 @@
+    <!-- /build -->
+
+    <script src="modules/dictation.js"></script>
+
+    <!-- build:css ../../css/bootstrap.min.css -->
+    <link href="../../html/css/bootstrap.min.css" rel="stylesheet">
+    <!-- /build -->
+    <!-- build:css ../../css/style.css -->
+    <link href="../../html/css/style.css" rel="stylesheet">
+    <!-- /build -->
+
+    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
+    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
+    <!--[if lt IE 9]>
+    <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
+    <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
+    <![endif]-->
+
+    <link rel="stylesheet" href="../common/css/style.css">
+</head>
+<body>
+<div class="container">
+    <div class="page-header">
+        <h1>Dictation Example</h1>
+        <p class="lead">
+            This is an example that uses a <code>Dictation</code> object to get text input into a Rocky app.
+        </p>
+    </div>
+    
+    <div class="col-md-6">
+        <canvas id="pebble" class="rocky" width="432" height="504"></canvas>    
+    </div>
+
+    <div class="col-md-6">
+        <h2>What's going on?</h2>
+        <p>
+            This example uses the Web Speech API to obtain dictated text from the user.
+        </p>
+        <p>
+            A <code>Dictation</code> object is created with <code>new Dictation();</code>. 
+        </p>
+        <p>
+            Next, the <code>start()</code> function is called and provided with two callbacks - <code>successCallback</code> that provides the transcribed text, and <code>failureCallback</code> that provides a string reason for any errors that occur.
+        </p>
+
+    </div>
+    <!-- build:template
+    <%= github_banner %>
+    /build -->
+</div>
+</body>
+
+<script>
+    (function () {
+        var rocky = Rocky.bindCanvas(document.getElementById("pebble"));
+        // puts all C-symbols from rocky onto window.* so they can be used almost
+        // the same way you were using them in C
+        rocky.export_global_c_symbols();
+
+        var lastTranscript = "Speak now";
+
+        rocky.update_proc = function (ctx, bounds) {
+            graphics_context_set_text_color(ctx, GColorBlack);
+            graphics_draw_text(ctx, lastTranscript, 
+                               fonts_get_system_font(FONT_KEY_GOTHIC_24_BOLD), bounds, 
+                               GTextOverflowModeWordWrap, GTextAlignmentCenter, null);
+        };
+
+        var lang = navigator.language;
+        var SpeechRecognition = SpeechRecognition || webkitSpeechRecognition;
+
+        var speechRecognition = new SpeechRecognition();
+        var dictation = new Dictation(speechRecognition, lang);
+        if (dictation) {
+            dictation.start(function (transcript) {
+                lastTranscript = "You said: \n" + transcript;
+                rocky.mark_dirty();
```

Comment:
if you pass in the `rocky` instance to `new Dictation()` the `mark_dirty()` could happen transparently.